vote-signer/src/rpc.rs

```diff
@@ -78,63 +77,117 @@ build_rpc_trait! {
         type Metadata;
 
         #[rpc(meta, name = "registerNode")]
-        fn register(&self, Self::Metadata, String) -> Result<Pubkey>;
+        fn register(&self, Self::Metadata, String, Signature, Vec<u8>) -> Result<Pubkey>;
 
         #[rpc(meta, name = "signVote")]
-        fn sign(&self, Self::Metadata, String) -> Result<Signature>;
+        fn sign(&self, Self::Metadata, String, Signature, Vec<u8>) -> Result<Signature>;
 
         #[rpc(meta, name = "deregisterNode")]
-        fn deregister(&self, Self::Metadata, String) -> Result<()>;
+        fn deregister(&self, Self::Metadata, String, Signature, Vec<u8>) -> Result<()>;
     }
 }
 
 pub struct VoteSignerRpcImpl;
 impl VoteSignerRpc for VoteSignerRpcImpl {
     type Metadata = Meta;
 
-    fn register(&self, meta: Self::Metadata, id: String) -> Result<Pubkey> {
+    fn register(
+        &self,
+        meta: Self::Metadata,
+        id: String,
+        sig: Signature,
+        signed_msg: Vec<u8>,
+    ) -> Result<Pubkey> {
         info!("register rpc request received: {:?}", id);
-        let pubkey = get_pubkey(id)?;
-        meta.request_processor.register(pubkey)
+        let pubkey = verify_pubkey(id)?;
+        if sig.verify(pubkey.as_ref(), signed_msg.as_ref()) {
```

Comment:
How about adding a utility function such that this and the 2 functions below this can be written without the if-else:

```rust
verify_signature(&sig, &pubkey, &signed_msg)?;
```

Reply:
👍